src/components/mapPageComponents/logic/event.ts

```diff
@@ -0,0 +1,190 @@
+import { predictPlace } from './predict';
+import { goToPlace, zoomToLocation } from './view';
+import { STREET_ZOOM } from '../constants';
+import { MapViews } from '../ducks/types';
+import {
+  setBlocksStyle,
+  setNeighborhoodsStyle,
+  setPrivateStreetsStyle,
+  setSitesStyle,
+} from './style';
+import { BasicTreeInfo, NO_TREE_PRESENT } from '../../treePopup';
+import { CheckboxValueType } from 'antd/es/checkbox/Group';
+
+// Logic for adding event listeners and handling events
+
+/**
+ * Adds an event listener to handle location searches.
+ * @param autocomplete the autocomplete to add the listener to
+ * @param autoService the autocomplete service to use
+ * @param placesService the places service to use
+ * @param callback the function to execute after predicting a place
+ * @param searchMarker the marker to display where the user has searched
+ * @param map the map
+ */
+export function addHandleSearch(
+  autocomplete: google.maps.places.Autocomplete,
+  autoService: google.maps.places.AutocompleteService,
+  placesService: google.maps.places.PlacesService,
+  callback: (
+    place: google.maps.places.PlaceResult,
+    status: google.maps.places.PlacesServiceStatus,
+  ) => void,
+  searchMarker: google.maps.Marker,
+  map: google.maps.Map,
+): void {
+  autocomplete.addListener('place_changed', () => {
+    const place: google.maps.places.PlaceResult = autocomplete.getPlace();
+    // If the place does not have a geometry (if the user did not enter a valid location)
+    if (!place.geometry) {
+      // Predicts the place the user wanted and sets the search marker at that place
+      predictPlace(place, autoService, placesService, callback);
+      return;
+    }
+
+    // Otherwise just goes to the place they searched for
+    goToPlace(place, searchMarker, map, STREET_ZOOM);
+  });
+}
+
+/**
+ * Adds an event listener to zoom to a neighborhood when it is clicked.
+ * @param neighborhoodsLayer the neighborhoods layer
+ * @param view the view
+ * @param map the map
+ */
+export function addZoomToClickedNeighborhood(
+  neighborhoodsLayer: google.maps.Data,
+  view: MapViews,
+  map: google.maps.Map,
+): void {
+  neighborhoodsLayer.addListener('click', (event) => {
+    zoomToLocation(
+      new google.maps.LatLng(
+        event.feature.getProperty('lat'),
+        event.feature.getProperty('lng'),
+      ),
+      map,
+      view,
+    );
+  });
+}
+
+// todo complete
```

Comment:
I would prefer if we removed commented out code but if we're going to get to this soon then its _ok_